tests/integ_tests/workflow_tests/timers.rs

```diff
@@ -1,30 +1,51 @@
-use assert_matches::assert_matches;
-use std::time::Duration;
+use std::{sync::Arc, time::Duration};
 use temporal_sdk_core::protos::coresdk::{
-    workflow_activation::{wf_activation_job, FireTimer, WfActivationJob},
     workflow_commands::{CancelTimer, CompleteWorkflowExecution, StartTimer},
     workflow_completion::WfActivationCompletion,
 };
-use temporal_sdk_core::test_workflow_driver::{CommandSender, TestRustWorker, TestWorkflowDriver};
+use temporal_sdk_core::test_workflow_driver::{CommandSender, TestRustWorker};
+use temporal_sdk_core::tracing_init;
 use test_utils::{init_core_and_create_wf, CoreWfStarter, NAMESPACE};
 
+async fn timer_wf(mut command_sink: CommandSender) {
+    let timer = StartTimer {
+        timer_id: "super_timer_id".to_string(),
+        start_to_fire_timeout: Some(Duration::from_secs(1).into()),
+    };
+    command_sink.timer(timer).await;
+    command_sink.complete_workflow_execution();
+}
+
 #[tokio::test(flavor = "multi_thread")]
-async fn timer_workflow_new_way() {
+async fn timer_workflow_not_sticky() {
```

Comment:
It would be nice to add a test that polls on sticky, fails or times out and then polls on non-sticky and is able to finish workflow using whatever was in the cache.

Reply:
That's a good idea will add

Reply:
Turns out this was already covered really, but, added a more specific test for it.